polytracker/src/polytracker/polytracker.cpp

```diff
@@ -10,22 +10,37 @@
 
 EARLY_CONSTRUCT_EXTERN_GETTER(taintdag::PolyTracker, polytracker_tdag);
 
+static bool polytracker_is_initialized = false;
```

Comment:
What is our approach to thread safety? I think we've attempted to keep things thread safe and due to that, we should probably change this into an [atomic_flag](https://en.cppreference.com/w/cpp/atomic/atomic_flag).

Reply:
Replaced with `std::atomic_flag`.